services/distribution/src/main/java/app/coronawarn/server/services/distribution/objectstore/ObjectStoreAccess.java

```diff
@@ -86,8 +89,10 @@
    *
    * @param localFile The file to be published.
    */
-  public void putObject(LocalFile localFile) {
+  @Async("s3TaskExecutor")
+  public Future<Void> putObject(LocalFile localFile) {
     putObject(localFile, DEFAULT_MAX_CACHE_AGE);
+    return new AsyncResult<>(null);
```

Comment:
looks weird - maybe the result of putObject should also be returned here

Reply:
s3client.putObject is a void method, so there is nothing to return here. We still need the Future to determine, whether the thread has finished or not.